src/main/java/org/mockito/internal/junit/WarningTestListener.java

```diff
@@ -0,0 +1,36 @@
+package org.mockito.internal.junit;
+
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.internal.util.MockitoLogger;
+
+import java.util.Collection;
+
+class WarningTestListener implements MockitoTestListener {
+
+    private final MockitoLogger logger;
+    private String testName;
+
+    WarningTestListener(MockitoLogger logger) {
+        this.logger = logger;
+    }
+
+    public void beforeTest(Object testClassInstance, String testMethodName) {
+        testName = testClassInstance.getClass().getSimpleName() + "." + testMethodName;
+        MockitoAnnotations.initMocks(testClassInstance);
+    }
+
+    public void afterTest(Collection<Object> mocks, Throwable testFailure) {
+        if (testFailure != null) {
+            //print stubbing mismatches only when there is a test failure
+            //to avoid false negatives. Give hint only when test fails.
+            new ArgMismatchFinder().getStubbingArgMismatches(mocks).format(testName, logger);
+        } else {
+            //Validate only when there is no test failure to avoid reporting multiple problems
+            Mockito.validateMockitoUsage();
```

Comment:
I wonder, if there's a test failure. But if some arguments are misplaced, users won't see the [`throw misplacedArgumentMatcher(lastMatchers)`](https://github.com/mockito/mockito/blob/v2.2.21/src/main/java/org/mockito/internal/progress/ArgumentMatcherStorageImpl.java#L102) ?